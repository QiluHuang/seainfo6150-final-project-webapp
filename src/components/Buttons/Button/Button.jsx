import React from 'react';
import PropTypes from "prop-types";
import styles from "../Button/Button.module.css";

const Button = (props) => {
    return (
        <button className={styles.button}>{props.buttonText}</button>
    )
}

Button.propTypes = {
    buttonText: PropTypes.string.isRequired,
    // onClick: PropTypes.func.isRequired,
}

export default Button;