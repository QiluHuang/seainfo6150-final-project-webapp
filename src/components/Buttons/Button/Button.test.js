import React from "react";
import { render } from "@testing-library/react";
import Button from "./Button.jsx";

const text = "Add";

describe("Button tests", () => {
    it("renders the Button component correctly", () => {
    const { container } = render(<Button buttonText={text} />);
    expect(container).toMatchSnapshot();
    });
});