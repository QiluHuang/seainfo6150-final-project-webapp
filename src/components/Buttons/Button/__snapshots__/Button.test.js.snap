// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders the Button component correctly 1`] = `
<div>
  <button
    class="button"
  />
</div>
`;
